extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnRunTimeConfig.java

```diff
@@ -0,0 +1,157 @@
+package io.quarkus.security.webauthn;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.vertx.ext.auth.webauthn.Attestation;
+import io.vertx.ext.auth.webauthn.AuthenticatorAttachment;
+import io.vertx.ext.auth.webauthn.AuthenticatorTransport;
+import io.vertx.ext.auth.webauthn.PublicKeyCredential;
+import io.vertx.ext.auth.webauthn.UserVerification;
+
+@ConfigRoot(name = "webauthn", phase = ConfigPhase.RUN_TIME)
+public class WebAuthnRunTimeConfig {
+
+    /**
+     * FIXME
+     */
+    @ConfigItem
+    public Optional<String> origin;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem
+    public Optional<String> domain;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "USB,NFC,BLE,INTERNAL")
+    public Optional<List<AuthenticatorTransport>> transports;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem
+    public RelyingPartyConfig relyingParty;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem
+    public Optional<AuthenticatorAttachment> authenticatorAttachment;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "false")
+    public Optional<Boolean> requireResidentKey;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "REQUIRED")
+    public Optional<UserVerification> userVerification;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "60s")
+    public Optional<Duration> timeout;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "NONE")
+    public Optional<Attestation> attestation;
+
+    // Needs to be a list, order is important
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "ES256,RS256")
+    public Optional<List<PublicKeyCredential>> pubKeyCredParams;
+
+    /**
+     * FIXME
+     */
+    @ConfigItem(defaultValueDocumentation = "64")
+    public OptionalInt challengeLength;
+
+    // FIXME
+    //private JsonObject extensions;
+
+    // FIXME
+    //private Map<String, X509Certificate> rootCertificates;
```

Comment:
You could enable this like in vert.x, you can use the "standard" base64 certificate encoder string:

https://github.com/vert-x3/vertx-auth/blob/master/vertx-auth-webauthn/src/main/java/io/vertx/ext/auth/webauthn/WebAuthnOptions.java#L47-L184

Reply:
I'll leave this open until someone wants this. I don't feel good telling people to add base64 certificates in the config file, that's just bad UX.
If anything, I'd prefer the config to point at files:

```properties
quarkus.webauthn.root-certificates.something=something.pem
```

Or even a convention where these files get automatically configured `src/main/resources/META-INF/webauthn/root/*.pem`

Reply:
BTW, what's the point of the `Map`? Why not just a list?

Reply:
The reason for the map was because older versions of vertx didn't supported all attestation modes and the root certificates are to be bound to a given attestation type. Internally vert.x uses a service loader to load the attestation implementations, so this was dynamic.

Perhaps we can improve the upstream code to not use a service loader as it now supports all attestations so we can define speicifc config properties, e.g.:

* android-safetynet
* android-key
* apple

And `mds` (metadata service).

We could have a dual option too: either inline buffer (which you don't like) and a filesystem path.

Reply:
We can handle files downstream, that's no problem. But I still don't understand why this is a `Map`, do you do anything special with the keys?

Reply:
yes, they need to match the attestation types:

* android-safetynet
* android-key
* apple

This way during the attestation we know which certificate to use.

Reply:
For reference, here's the service loader config:

https://github.com/vert-x3/vertx-auth/blob/master/vertx-auth-webauthn/src/main/resources/META-INF/services/io.vertx.ext.auth.webauthn.impl.attestation.Attestation

And the names are coming from the interface:

https://github.com/vert-x3/vertx-auth/blob/master/vertx-auth-webauthn/src/main/java/io/vertx/ext/auth/webauthn/impl/attestation/Attestation.java#L37-L41